api/src/main/java/org/pmiops/workbench/api/OfflineUserController.java

```diff
@@ -84,39 +78,16 @@ public OfflineUserController(
         Timestamp newTime = updatedUser.getComplianceTrainingCompletionTime();
         DataAccessLevel newLevel = updatedUser.getDataAccessLevelEnum();
 
-        if (!Objects.equals(newTime, oldTime)) {
-          log.info(
-              String.format(
-                  "Compliance training completion changed for user %s. Old %s, new %s",
-                  user.getUsername(), oldTime, newTime));
-          changeCount++;
-        }
-        if (oldLevel != newLevel) {
-          log.info(
-              String.format(
-                  "Data access level changed for user %s. Old %s, new %s",
-                  user.getUsername(), oldLevel.toString(), newLevel.toString()));
-          accessLevelChangeCount++;
-        }
+        completionChangeCount += logCompletionChange(user, oldTime, newTime, "Compliance training");
```

Comment:
I think it might be nice to create an enum to capture the different module types. Easier to change string values and it's better for type safety to pass that around than strings.